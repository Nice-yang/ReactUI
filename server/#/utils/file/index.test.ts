import { readFileToString$ } from './index';

it('should give string of a text file', done => {
  readFileToString$(`${__dirname}/index.ts`).subscribe(text => {
    expect(typeof text).toMatch('string');
    done();
  });
});

it('should not work with something which is not a file', done => {
  readFileToString$(__dirname).subscribe({
    error: x => {
      expect(x).toMatchSnapshot();
      done();
    },
  });
});
