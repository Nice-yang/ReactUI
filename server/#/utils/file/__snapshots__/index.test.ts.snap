// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`getPathRelativeToTarget should resolve .. paths 1`] = `"../../../index.ts"`;

exports[`getPathRelativeToTarget should resolve ./ paths 1`] = `"./target1.ts"`;

exports[`getPathRelativeToTarget should throw for invalid paths 1`] = `
Object {
  "data": [Error: Cannot find module './targe1' from '/Users/I0459/scratch/personal/ellipsoid/server/#/utils/file/__fixtures__'],
  "level": 0,
  "message": "Could not get path relative to target",
  "stack": undefined,
}
`;

exports[`readFileToString$ should not work with something which is not a file 1`] = `
Object {
  "data": Object {
    "error": "{\\"errno\\":-21,\\"code\\":\\"EISDIR\\",\\"syscall\\":\\"read\\"}",
    "fileName": "/Users/I0459/scratch/personal/ellipsoid/server/#/utils/file",
  },
  "level": 0,
  "message": "Cannot read file",
  "stack": undefined,
}
`;
